source/agora/node/admin/QRCodeInterface.d

```diff
@@ -0,0 +1,289 @@
+        Copyright (c) 2020 BOS Platform Foundation Korea
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+module agora.node.admin.QRCodeInterface;
+
+import agora.common.Config;
+import agora.common.crypto.Key;
+import agora.common.Hash;
+import agora.common.Types;
+import agora.consensus.data.PreImageInfo;
+import agora.utils.Log;
+import agora.network.Clock;
+
+import barcode;
+
+import vibe.data.json;
+import vibe.http.router;
+
+import core.stdc.time;
+import std.datetime : SysTime, unixTimeToStdTime;
+import std.datetime.timezone : UTC;
+import std.format;
+
+import core.time : days;
+
+mixin AddLogger!();
+
+/*******************************************************************************
+
+    Implementation of the Agora User's credentials QR services
+
+    A class to provide a login or encryption key to QRs to verify that
+    the owner of the validator from outside agora.
+    Respond SVG format's QR according to the get requests.
+    Contains QR code generator.
+
+*******************************************************************************/
+
+public class QRCodeInterface
+{
+    /// The keypair of this node
+    private const KeyPair key_pair;
+
+    /// Clock instance
+    private Clock clock;
+
+    /***************************************************************************
+
+        Constructor
+
+        Params:
+            key_pair = the keypair of this node
+
+    ***************************************************************************/
+
+    public this (const KeyPair key_pair, Clock clock)
+    {
+        this.key_pair = key_pair;
+        this.clock = clock;
+    }
+
+    /***************************************************************************
+
+        Respond with QR code containing login information.
+
+        Enter the name of the APP where the login information will be
+        used as the get query.
+
+        Convert LoginInfo to JSON format and respond with the QR code
+        of SVG format.
+        All public key cryptography is done on Curve25519.
+        The signature of the vectorCard provides verification function in the SDK.
+        This LoginInfo has an expiration time.
+        It provide a randomly generated temporary private key encoded
+        in strings on demand.
+        In the future, there is a need to enter a temporary key.
+
+    ***************************************************************************/
+
+    public void loginQRCreate (scope HTTPServerRequest req,
+        scope HTTPServerResponse res)
+    {
+        // The randomly generated temporary KeyPair
+        KeyPair temp_kp = KeyPair.random();
+
+        time_t current_time = this.clock.networkTime();
+        // A 'nonce' used to allow expiring
+        current_time += 60 * 60 * 24 * 90; // (default: +90 days)
+        SysTime expires_time = SysTime(unixTimeToStdTime(current_time));
```

Comment:
I just ran it and I was able to get this.
expores_time: "2021-01-28T10:24:06"
It's working fine.